test/type/opaque.swift

```diff
@@ -0,0 +1,225 @@
+// RUN: %target-swift-frontend -typecheck -verify -enable-opaque-result-types %s
+
+// TODO: The syntax is only parsed, and opaque decls are synthesized for 
+// function decls, but the opaque types are not themselves formed, resolved,
+// or checked yet.
+
+protocol P {
+  func paul()
+  mutating func priscilla()
+}
+protocol Q { func quinn() }
+extension Int: P, Q { func paul() {}; mutating func priscilla() {}; func quinn() {} }
+extension String: P, Q { func paul() {}; mutating func priscilla() {}; func quinn() {} }
+extension Array: P, Q { func paul() {}; mutating func priscilla() {}; func quinn() {} }
+
+class C {}
+class D: C, P, Q { func paul() {}; func priscilla() {}; func quinn() {} }
+
+// TODO: Should be valid
+
+let foo: __opaque P = 1 // FIXME expected-error{{'opaque' types are only implemented}}
+var computedFoo: __opaque P {  // FIXME expected-error{{'opaque' types are only implemented}}
```

Comment:
What about subscripts?

And a read-only computed property (or subscript) should be allowed, no?

Reply:
Yeah, I was planning to add support for read-only properties and subscripts too.